app/controllers/admin/reports_controller.rb

```diff
@@ -4,6 +4,14 @@ module Admin
   class ReportsController < Spree::Admin::BaseController
     include ActiveStorage::SetCurrent
     include ReportsActions
+
+    # Time to wait for a background report to be generated.
+    # This value needs to be smaller than the `puma_timeout` in ofn-install
+    # which sets `ENV["RACK_TIMEOUT_SERVICE_TIMEOUT"]` and nginx timeouts.
+    # But it needs to be larger than ReportJob::NOTIFICATION_TIME or we may
+    # time out without sending a notification email to download the report.
+    MAX_WAIT_TIME = 10.seconds
```

Comment:
I was wondering about this.
Yes, some servers have a long timeout, but that was only because of reports, which this PR will resolve. Then those timeouts could be adjusted to be more normal (eg 1-2min).

So why not just use the full server timeout period (less 2 secs), which the instance manager may configure to suit their needs? This would allow a better UX by providing an on-screen response rather than email.

But this change frees up workers from unnecessary waiting, which has a performance benefit. Maybe this value should be configurable so that instance managers can determine the balance between UX and performance.

But.. with a reflex we can solve the UX problem..

Reply:
> with a reflex we can solve the UX problem..

Exactly, that was my conclusion. This change is to give quicker feedback to the user and free up workers. But I take your point. We could delay this change until we have the UX problem solved.

I'm not happy with the "full server timeout period (less 2 secs)" though because the two seconds are a guess. That may be more of a theoretical problem though.

Once we have a reflex, maybe we don't want to wait at all. We could display a "loading" screen straight away.

Okay, I think you convinced me that I should revert this change. We'll solve it with the reflex. Low number of workers has not been a reported problem and if that's because of too many reports then it may be good to limit that...

Reply:
Haha sorry I didn't meant to convince you to change it now. But yes I had the same conclusion: if we can push a response on completion with a reflex, then we don't need to wait here at all.